lib/gollum-lib/filter/yaml.rb

```diff
@@ -0,0 +1,26 @@
+require 'yaml'
+# Extract YAML frontmatter from data and build metadata table. 
+
+class Gollum::Filter::YAML < Gollum::Filter
+  
+  # Regexp thanks to jekyll
+  YAML_FRONT_MATTER_REGEXP = %r!\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)!m
+
+  def extract(data)
+    data.gsub!(YAML_FRONT_MATTER_REGEXP) do
+      @markup.metadata ||= {}
+      begin
+        frontmatter = ::YAML.load(Regexp.last_match[1])
+        frontmatter = frontmatter.respond_to?(:to_h) ? frontmatter.to_h : Hash.new(frontmatter)
+        @markup.metadata.merge!(frontmatter)
+      rescue Psych::SyntaxError
```

Comment:
Yes, good idea. An `errors` keyword with an `Array` as value? That could actually be a place to store errors collected in the rest of the filter chain, too, should we want to.